### PR TITLE
feat(satellite): improvements to the subscription proto

### PR DESCRIPTION
**Pull request by balegas**
* handle FK constraints in gc/applying txn
* send unsubscribe messages on error
* map proto errors to sat errors

### CODE REVIEW

**Review by kevin-dp — APPROVED** [review]

File: clients/typescript/src/satellite/mock.ts [hunk 1]
```diff
@@ -124,58 +131,67 @@ export class MockSatelliteClient extends EventEmitter implements Client {
...
-    const subscriptionId = randomValue()
-
-    const tablename = shapes[0]?.definition.selects[0]?.tablename
-    if (tablename == 'parent' || tablename == 'child') {
-      const shapeReqToUuid = {
-        [shapes[0].requestId]: uuid(),
-      }
+  setRelationData(tablename: string, record: DataRecord): void {
+    let relationData: DataRecord[]
+    if ((relationData = this.relationData[tablename] ?? [])) {
```
[the remark this answers is not in the document]
Comment (balegas) [reply clients/typescript/src/satellite/mock.ts]: Fixed that code that I messed up trying doing it in a more idiomatic way

File: clients/typescript/src/satellite/process.ts [hunk 2]
```diff
@@ -227,18 +227,25 @@ export class SatelliteProcess implements Satellite {
...
-        },
-        ...this._enableTriggers([tablename])
-      )
+  async _garbageCollectShapeHandler(
+    shapeDefs: ShapeDefinition[]
+  ): Promise<void> {
+    const stmts: Statement[] = []
+    // reverts to off on commit/abort
+    stmts.push({ sql: 'PRAGMA defer_foreign_keys = ON' })
+    for (const shapeDef of shapeDefs) {
```
[the remark this answers is not in the document]
Comment (balegas) [reply clients/typescript/src/satellite/process.ts]: That is a good suggestion.

**Review by kevin-dp — APPROVED** [review]

**Review by balegas — COMMENTED** [review]
Addressed suggestions and left some comments

File: clients/typescript/src/satellite/shapes/types.ts [hunk 3]
```diff
@@ -10,6 +10,8 @@ export type SubscribeResponse = {
   subscriptionId: string
 }
 
+export type UnsubscribeResponse = Record<string, never>
```
[the remark this answers is not in the document]
Comment (kevin-dp) [reply clients/typescript/src/satellite/shapes/types.ts:13]: omg, only TS things.. ;D

File: clients/typescript/src/satellite/client.ts [hunk 4]
```diff
@@ -169,6 +174,10 @@ export class SatelliteClient extends EventEmitter implements Client {
           handle: (msg: SatShapeDataEnd) => this.handleShapeDataEnd(msg),
           isRpc: false,
         },
+        SatUnsubsResp: {
+          handle: (_msg: SatUnsubsResp) => ({}),
```
[the remark this answers is not in the document]
Comment (kevin-dp) [reply clients/typescript/src/satellite/client.ts]: Thanks! Will be helpful later when i will implement unsubscribe in the DAL.

**Pull request merged**